src/databricks/labs/ucx/assessment/dashboards.py

```diff
@@ -0,0 +1,370 @@
+from __future__ import annotations
+
+import itertools
+import json
+import logging
+from collections.abc import Iterable, Iterator
+from dataclasses import dataclass, field
+
+from databricks.labs.lsql.backends import SqlBackend
+from databricks.labs.lsql.lakeview import Dashboard as LsqlLakeviewDashboard, Dataset
+from databricks.sdk import WorkspaceClient
+from databricks.sdk.errors import DatabricksError
+from databricks.sdk.service.dashboards import Dashboard as SdkLakeviewDashboard
+from databricks.sdk.service.sql import Dashboard as SdkRedashDashboard, LegacyQuery
+
+from databricks.labs.ucx.framework.crawlers import CrawlerBase
+from databricks.labs.ucx.framework.utils import escape_sql_identifier
+
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class Query:
+    """UCX representation of a Query.
+
+    Note:
+        This class is not persisted into an inventory table. If you decide to persist this class, consider (future)
+        differences between Redash and Lakeview queries
+    """
+
+    id: str
+    """The ID for this query."""
+
+    name: str = "UNKNOWN"
+    """The title of this query that appears in list views, widget headings, and on the query page."""
+
+    parent: str = "ORPHAN"
+    """The identifier of the workspace folder containing the object."""
+
+    query: str = ""
+    """The text of the query to be run."""
+
+    catalog: str = ""
+    """The name of the catalog to execute this query in."""
+
+    schema: str = ""
+    """The name of the schema to execute this query in."""
```

Comment:
I don't think we should have default values here: it's an error if they have not been set, and assigning them a (nonsense) default masks errors because we end up persisting nonsensical data.